### PR TITLE
Update top level readme with links to more information

### PR DESCRIPTION
**Pull request by andrewhop**
### Issues:
Resolves CryptoAlg-733

### Description of changes: 
Update the Readme to accurately reflect the state of the project and include useful links to more information.

### Testing:
It is easier to review a rendered version of the readme on my repository https://github.com/andrewhop/aws-lc/tree/readme_updates#readme

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by darylmartin100 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,28 +1,130 @@
-# AWS libcrypto (AWS-LC) Public Preview
+# AWS libcrypto (AWS-LC)
 
 [![Join the chat at https://gitter.im/awslabs/aws-lc](https://badges.gitter.im/awslabs/aws-lc.svg)](https://gitter.im/awslabs/aws-lc?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
-AWS libcrypto (AWS-LC) is a general-purpose cryptographic library maintained by
-the AWS Cryptography team for use with AWS Nitro Enclaves and other AWS applications. 
-AWS-LC іs based on code from the Google BoringSSL project and the OpenSSL project.
+AWS-LC is a general-purpose cryptographic library maintained by the AWS Cryptography
+team for AWS and there customers.It іs based on code from the Google BoringSSL project
```
Comment (samuel40791765) [inline README.md:6]: Nit: their customers. It

File: README.md [hunk 2]
```diff
@@ -1,28 +1,130 @@
...
+standard. To build libssl and AWS-LC’s tests you will need a compiler that supports
+C++11. AWS-LC is tested with:
+
+* GCC 4.8.5 to 8.4
+* Clang 7 to 10
+* Visual Studio 2015
+
+For a complete list of tested compilers see
+[tests/ci/Readme.md](https://github.com/awslabs/aws-lc/blob/main/tests/ci/README.md).
+If you use another compiler and would like to make sure we maintain support please
```
Comment (samuel40791765) [inline README.md:62]: Tiny nit: "," before please. A couple below as well.

File: README.md [hunk 3]
```diff
@@ -1,28 +1,130 @@
...
+If you have any questions about Submitting PR's, Opening Issues, AWS-LC API usage or
+any similar topic, we have a public chatroom available here to answer your questions
+on [Gitter](https://gitter.im/awslabs/aws-lc).
+
+Otherwise, if you think you might have found a security impacting issue, please instead
+follow our Security Notification Process.
 
 ## Security issue notifications
+
 If you discover a potential security issue in AWS-LC we ask that you notify AWS
```
Comment (samuel40791765) [inline README.md:25]: Tiny nit: comma before we

File: README.md [hunk 4]
```diff
@@ -1,28 +1,130 @@
...
+[Contributing.md](https://github.com/awslabs/aws-lc/blob/main/CONTRIBUTING.md) has
+info for how to specifically make the change and get it reviewed by AWS-LC maintainers.
+If you just want to use AWS-LC see our existing documentation in the public header
+files, if you’re moving your application from OpenSSL see
+[Porting_to_AWS-LC.md](https://github.com/awslabs/aws-lc/blob/main/PORTING_TO_AWSLC.md)
+for more information.
 
 ## Why AWS-LC?
 
 AWS-LCs goal is to maintain a secure libcrypto that is compatible with software and
```
Comment (samuel40791765) [inline README.md:11]: Nit: AWS-LC's

File: README.md [hunk 5]
```diff
@@ -1,28 +1,130 @@
...
 
-AWS libcrypto (AWS-LC) is a general-purpose cryptographic library maintained by
-the AWS Cryptography team for use with AWS Nitro Enclaves and other AWS applications. 
-AWS-LC іs based on code from the Google BoringSSL project and the OpenSSL project.
+AWS-LC is a general-purpose cryptographic library maintained by the AWS Cryptography
+team for AWS and there customers.It іs based on code from the Google BoringSSL project
+and the OpenSSL project.
+
+AWS-LC contains portable C implementations of algorithms needed for TLS and common
+applications. For performance critical algorithms, optimized assembly versions are
```
Comment (samuel40791765) [inline README.md:10]: Do we want to include which algorithms we have optimized assembly versions for?

**Review by samuel40791765 — COMMENTED** [review]
Just some tiny nits and a question

File: README.md (hunk 5 above)
Comment (andrewhop) [reply README.md:10]: I want to include that change with the performance updates change.

**Review by samuel40791765 — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1,28 +1,130 @@
...
+
+### OS Compatibility
+
+AWS-LC should work on any system with a supported compiler. AWS-LC has been written to
+support Unix and Windows operating systems. AWS-LC is tested on:
+* Amazon Linux 2
+* Centos 7
+* Ubuntu 18.04 and 20.04
+* Fedora 31
+* Windows Server 10
```
Comment (bryce-shang) [inline README.md:73]: Nit: the Docker file uses `FROM mcr.microsoft.com/windows/servercore:ltsc2019`. Is Windows 10 another name of windows server 2019? I saw CodeBuild uses Windows 2019 directly.

**Review by bryce-shang — APPROVED** [review]

**Pull request merged**